### PR TITLE
Auto subscribe

### PR DESCRIPTION
**Pull request by kimyen**
When a project is created, create the forum and subscribe the project owner to the forum.

When a user being added as admin for a project, we may decide to send a notification to the user and ask them to subscribe to the forum. 

When a user subscribe to a forum, he/she will also be subscribed to all existing threads in the forum.

When a thread is created, subscribe all forum subscribers to this thread.

Add a new API to get the subscription given its ID.
Add a new API to get forum metadata given its ID.

### CODE REVIEW

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java [hunk 1]
```diff
@@ -75,17 +83,29 @@ public EntityManagerImpl(NodeManager nodeManager,
...
+		EntityType type = EntityTypeUtils.getEntityTypeForClass(newEntity.getClass());
+		node.setNodeType(type);
 		node.setActivityId(activityId);
 		NamedAnnotations annos = new NamedAnnotations();
 		// Now add all of the annotations and references from the entity
 		NodeTranslationUtils.updateNodeSecondaryFieldsFromObject(newEntity, annos.getPrimaryAnnotations());
 		// We are ready to create this node
 		String nodeId = nodeManager.createNewNode(node, annos, userInfo);
+		if (type == EntityType.project) {
+			createForumAndSubscribe(userInfo, nodeId);
```
Comment (kimyen) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java]: move the logic to ProjectMetadataProvider, use the Service to create project

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java [hunk 2]
```diff
@@ -75,17 +83,29 @@ public EntityManagerImpl(NodeManager nodeManager,
...
+		}
 		// Return the id of the newly created entity
 		return nodeId;
 	}
 
+	private void createForumAndSubscribe(UserInfo userInfo, String nodeId) {
```
Comment (kimyen) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/EntityManagerImpl.java]: Move this method to SubscriptionManager

File: services/repository/src/main/java/org/sagebionetworks/repo/web/service/metadata/TypeSpecificCreateProvider.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package org.sagebionetworks.repo.web.service.metadata;
+
+import org.sagebionetworks.repo.model.Entity;
+import org.sagebionetworks.repo.model.UserInfo;
+
+public interface TypeSpecificCreateProvider<T extends Entity> extends EntityProvider<T> {
```
Comment (kimyen) [inline services/repository/src/main/java/org/sagebionetworks/repo/web/service/metadata/TypeSpecificCreateProvider.java:6]: rename to AfterCreate

File: client/synapseJavaClient/src/main/java/org/sagebionetworks/client/SynapseClient.java [hunk 4]
```diff
@@ -2659,4 +2668,12 @@ void updateVerificationState(long verificationId,
...
 	 */
 	void unsubscribeAll() throws SynapseException;
+
+	/**
+	 * Retrieve a subscription given its ID
+	 * 
+	 * @param subscriptionId
+	 * @throws SynapseException 
+	 */
+	Subscription get(String subscriptionId) throws SynapseException;
```
Comment (jay-hodgson) [inline client/synapseJavaClient/src/main/java/org/sagebionetworks/client/SynapseClient.java:2678]: Prefer a different name, like getSubscription().

**Pull request merged**